src/NuGet.Core/NuGet.Protocol/Plugins/IConnection.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Versioning;
+
+namespace NuGet.Protocol.Plugins
+{
+    /// <summary>
+    /// Represents a bidirectional channel between a NuGet client and a plugin.
+    /// </summary>
+    public interface IConnection : IDisposable
```

Comment:
There isn't a public scenario for these methods.

We'd like to reuse plugin instances to minimize the startup cost, so each plugin instance is a singleton.

This is an implementation detail and clients aren't aware of it.  Clients can multiplex on the same plugin instance.  `PluginFactory` manages singleton lifetime; on idle, a plugin instance is disposed.

It would be tempting for a client to call `ConnectAsync()` or `CloseAsync()`, but that would break other clients using the same instance.

`PluginFactory` calls `ConnectAsync()` and `CloseAsync()`, as appropriate.